### ISSUE TITLE
Errors in implementation of K0->ll'

### ISSUE THREAD

**Issue by hoodyn** [flav-io/flavio#138]
Hi, 

I think I've found two problems in the implementation of K0->ll' in flavio/physics/kdecays/kll.py.

### Lepton flavour violating K-short decay results in errors
`K=KS` together with `l1 != l2` not a valid option in the definition of function `amplitudes_eff`:
```
   ...
    if K == 'KS' and l1 == l2:
        Peff = P.imag
        Seff = S.real + SLD
    if K == 'KL':
        Peff = P.real + PLD
        Seff = S.imag
    return Peff, Seff
```
Correspondingly, asking flavio for a prediction on `BR(KS->emu,mue)` leads to 
`UnboundLocalError: local variable 'Peff' referenced before assignment`.

This would have been trivial to fix, be there not the second problem. 

### Lepton flavour violating K-long decay calculated incorrectly
This is physIcs-related issue: the formulas for using imaginary of real parts of the Wilson coefficients (or, of the `P,S` parts of the `amplitudes` in the current flavio implementation) hold only for the lepton flavour conserving case. 

Generally, KL and KS are superpositions of K0 and antiK0: K_L,S = (sbar d +- dbar s) / sqrt(2). 
Thus, naively, the amplitudes consist of the combinations `(CX_sdl1l2 +- CX_dsl1l2)/sqrt(2)` with X=9,9p,10,10p,S,SP,P,Pp.
However, in flavio, the `_dsl1l2` Wilson coefficients are not to be specified as they are obtained from complex-conjugated `_sdl2l1` ones. In particular: 
![image](https://user-images.githubusercontent.com/77328256/105366951-33939f00-5c00-11eb-9022-cd777b39f038.png)
(The extra minus sign for `CS-CSp` is spit by the CP transformation of the corresponding operator - I would appreciate if someone could check the above formulas independently. The  `CX + CXp` combinations are irelevant for pseudoscalar meson decays.)
If `l1==l2`, one gets terms like `( (CX_sdll-CXp_sdll) +- (CX_sdll-CXp_sdll).conj() ) / sqrt(2)` which simplify to real or i*imaginary parts indeed.
However, the LFV cases lead to expressions like `( (CX_sdemu-CXp_sdemu) +- (CX_sdmue-CXp_sdmue).conj() ) / sqrt(2)` which obviously cannot be simplified any further.

Regards, Matej Hudec

**Comment by hoodyn** [comment]
If you agree, I would volunteer to do the repair, but I'm not sure about the policy: shall I fork the project, or will you give me the rights to set up a branch?

NB: in the formulas above, _xi_t = V[t,s] V[t,d]*_ is the complex factor in the normalization coefficient of the effective operators.

**Comment by DavidMStraub** [comment]
Hi,

while I am responsible for the implementation and it's definitely possible that your formulas are correct, unfortunately I don't have time to look into this in detail, so someone else would have to check it.

Concerning the repair: yes, it works with forking and pull requests, as usual.

**Comment by peterstangl** [comment]
@hoodyn, I agree that the implementation has the two issues you have described. Thank you for reporting them and for your offer to fix them!

For implementing the modifications, you should fork the repository, create a new branch in the forked repository, commit your changes to this new branch, and then go to the main flavio repository on the GitHub website were you will be able to open a "Pull Request" (PR). Once you have opened the PR, I will check the implementation before merging it into the main flavio repository.

**Comment by peterstangl** [comment]
@hoodyn, yes, I actually went through most of your implementation and the derivation in the notes you provided (which was very helpful!). I noticed some issues in the implementation of the long-distant part but I had no time yet to completely finish all checks. I will comment on what I found so far in the PR as soon as I have time to look into this again (hopefully within the next couple of weeks).
